include/dd/Package.hpp

```diff
@@ -2153,6 +2178,63 @@ namespace dd {
             return r;
         }
 
+        std::map<std::string, double> getProbVectorFromDensityMatrix(dEdge e, double measurementThreshold) {
+            std::map<std::string, double> measuredResult = {};
+            double                        p0, p1;
+            double                        globalProbability;
+            double                        statesToMeasure;
+
+            dEdge::alignDensityEdge(&e);
+
+            statesToMeasure = std::pow(2, nqubits);
+
+            for (int m = 0; m < statesToMeasure; m++) {
```

Comment:
`int` certainly seems the wrong type for the iteration here... there are cases, where this will result in an endless loop. Specifically, any density matrix with more than `32` qubits will never exit this loop.
Probably best to use a fixed width **unsigned** type here and add a check whether the number of qubits in the state does not exceed the bitwidth of that fixed width type.

Correspondingly adapt the types below.

Reply:
Increased the size of the involved variables to unsigned long long and added a check to make sure that no overflow occurs.